api/src/main/java/com/epam/esm/web/handler/RestExceptionHandler.java

```diff
@@ -0,0 +1,226 @@
+        String message = messageManager.receiveMessage(NEWLY_CREATED_ABSENCE_PROPERTY, locale);
+        ErrorInfo errorInfo = new ErrorInfo(message, ErrorCode.NEWLY_CREATED_ABSENCE);
+        return new ResponseEntity<>(errorInfo, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<ErrorInfo> illegalArgumentsHandle(IllegalArgumentException e, Locale locale) {
+        String message = messageManager.receiveMessage(INVALID_ARGUMENT_PROPERTY, locale);
+        String errorMessage = message + " (" + e.getMessage() + ")";
+        ErrorInfo errorInfo = new ErrorInfo(errorMessage, ErrorCode.ILLEGAL_ARGUMENT);
+        return new ResponseEntity<>(errorInfo, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler({NotExistentUpdateException.class})
+    public ResponseEntity<ErrorInfo> notExistingUpdateHandle(NotExistentUpdateException e, Locale locale) {
+        String message = messageManager.receiveMessage(NON_EXISTING_UPDATE_PROPERTY, locale);
+        String errorMessage = message + " (id = " + e.getId() + ")";
+        ErrorInfo errorInfo = new ErrorInfo(errorMessage, ErrorCode.NOT_EXISTENT_UPDATE);
+        return new ResponseEntity<>(errorInfo, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler({HttpRequestMethodNotSupportedException.class})
+    public ResponseEntity<?> unsupportedMethodHandling() {
+        HttpHeaders headers = new HttpHeaders();
+        Set<HttpMethod> allowedMethods = new HashSet<>();
+        allowedMethods.add(HttpMethod.GET);
+        allowedMethods.add(HttpMethod.PUT);
+        allowedMethods.add(HttpMethod.DELETE);
+        allowedMethods.add(HttpMethod.POST);
+        headers.setAllow(allowedMethods);
+        return new ResponseEntity<>(headers, HttpStatus.METHOD_NOT_ALLOWED);
+    }
+
+    @ExceptionHandler(TagDuplicateException.class)
+    public ResponseEntity<ErrorInfo> tagDuplicateCreatingHandle(TagDuplicateException e, Locale locale) {
+        String message = messageManager.receiveMessage(TAG_DUPLICATE_PROPERTY, locale);
+        String errorMessage = message + " (" + e.getMessage() + ")";
+        ErrorInfo errorInfo = new ErrorInfo(errorMessage, ErrorCode.CREATE_TAG_DUPLICATE);
+        return new ResponseEntity<>(errorInfo, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(GiftCertificateDuplicateException.class)
+    public ResponseEntity<ErrorInfo> certificateDuplicateCreatingHandle(GiftCertificateDuplicateException e,
+                                                                        Locale locale) {
+        String message = messageManager.receiveMessage(CERTIFICATE_DUPLICATE_PROPERTY, locale);
+        String errorMessage = message + " (" + e.getMessage() + ")";
+        ErrorInfo errorInfo = new ErrorInfo(errorMessage, ErrorCode.CREATE_CERTIFICATE_DUPLICATE);
+        return new ResponseEntity<>(errorInfo, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
```

Comment:
This exception handler won't work, since it's required to override `ResponseEntityExceptionHandler`s method to make it work.